### ISSUE TITLE
Workspaces starting with some missing things

### ISSUE THREAD

**Issue by trumbitta** [gitpod-io/gitpod#8569]
### Bug description

Details: https://discord.com/channels/816244985187008514/816246578594840586/948167113187659808

(copy-pasting from Discord)

workspace opened with no Gitpod extension? (there's no Gitpod: open in VSCode item in the command palette)
Not the first time this happened: opening a new workspace usually fixes it
If the workspace id is the first part of the URL minus gitpod.io , I'd rather not share it because it has parts of our internal repo structure in it
Reloading the page fixed it

```
refreshing search data
/_supervisor/v1/status/supervisor:1 
        
      
        
        
      
        
      
       
        
       Failed to load resource: the server responded with a status of 502 ()
iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io/:1 
        
       Access to manifest at 'https://blobserve.ws-eu34.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-890de80174c7dd9d073d63ef2b1f9eac68f063c1/__files__/manifest.json' (redirected from 'https://iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io/manifest.json') from origin 'https://iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
blobserve.ws-eu34.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-890de80174c7dd9d073d63ef2b1f9eac68f063c1/__files__/manifest.json:1 
        
      
        
        
      
        
      
       
        
       Failed to load resource: net::ERR_FAILED
/_supervisor/v1/status/supervisor:1 
        
      
        
        
      
        
      
       
        
       Failed to load resource: the server responded with a status of 502 ()
/_supervisor/v1/status/supervisor:1 
        
      
        
        
      
        
      
       
        
       Failed to load resource: the server responded with a status of 502 ()
/_supervisor/v1/status/supervisor:1 
        
      
        
        
      
        
      
       
        
       Failed to load resource: the server responded with a status of 502 ()
reconnecting-websocket-mjs.js:517 
        
       WebSocket connection to 'ws://localhost:63100/localapp.LocalApp/TunnelStatus' failed: 
(anonymous) @ reconnecting-websocket-mjs.js:517
workbench.ts:774 
        
       cannot maintain connection to local app Object
(anonymous) @ workbench.ts:774
reconnecting-websocket-mjs.js:517 
        
       WebSocket connection to 'ws://localhost:63100/localapp.LocalApp/TunnelStatus' failed: 
(anonymous) @ reconnecting-websocket-mjs.js:517
workbench.ts:774 
        
       cannot maintain connection to local app Object
(anonymous) @ workbench.ts:774
reconnecting-websocket-mjs.js:517 
        
       WebSocket connection to 'ws://localhost:63100/localapp.LocalApp/TunnelStatus' failed: 
(anonymous) @ reconnecting-websocket-mjs.js:517
workbench.ts:774 
        
       cannot maintain connection to local app Object
(anonymous) @ workbench.ts:774
log.ts:313   ERR ENOENT: no such file or directory, access '/workspace/.vscode-remote/extensions/dbaeumer.vscode-eslint-2.2.2': Error: ENOENT: no such file or directory, access '/workspace/.vscode-remote/extensions/dbaeumer.vscode-eslint-2.2.2'
2abstractExtensionService.ts:303 
        
       Uncaught (in promise) Error: ENOENT: no such file or directory, access '/workspace/.vscode-remote/extensions/dbaeumer.vscode-eslint-2.2.2'
iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io/:1 
        
       Access to internal resource at 'https://blobserve.ws-eu34.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-890de80174c7dd9d073d63ef2b1f9eac68f063c1/__files__/manifest.json' (redirected from 'https://iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io/manifest.json') from origin 'https://iungoproducts-iungofront-qtxcx0hsc3t.ws-eu34.gitpod.io' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'.
DevTools failed to load source map: Could not load content for https://blobserve.ws-eu34.gitpod.io/eu.gcr.io/gitpod-core-dev/build/ide/code:commit-890de80174c7dd9d073d63ef2b1f9eac68f063c1/__files__/node_modules/@gitpod/local-app-api-grpcweb/lib/localapp.js.map: HTTP error: status code 404, net::ERR_HTTP_RESPONSE_CODE_FAILURE
reconnecting-websocket-mjs.js:517 
        
       WebSocket connection to 'ws://localhost:63100/localapp.LocalApp/TunnelStatus' failed: 
(anonymous) @ reconnecting-websocket-mjs.js:517
workbench.ts:774 
        
       cannot maintain connection to local app {code: 2, details: 'Response closed without headers', metadata: e}
(anonymous) @ workbench.ts:774
reconnecting-websocket-mjs.js:517 
        
       WebSocket connection to 'ws://localhost:63100/localapp.LocalApp/TunnelStatus' failed: 
(anonymous) @ reconnecting-websocket-mjs.js:517
workbench.ts:774 
        
       cannot maintain connection to local app {code: 2, details: 'Response closed without headers', metadata: e}
(anonymous) @ workbench.ts:774
workbench.ts:715 
        
       could not connect to local app 5 times, giving up, use 'Gitpod: Connect to Local App' command to retry
(anonymous) @ workbench.ts:715
```

it's happening with 99% of my workspaces now
another common symptom is that the terminal won't automatically open
these are workspaces of a project with (incremental) prebuilds, if that matters
reloading the page still fixes it all 

ok make it 100%, they're also all coming from GitLab merge requests
this is my network 

![](https://www.speedtest.net/result/12841368513.png)

### Steps to reproduce

Open a workspace from a GitLab MR (this is just where I'm experiencing it at the time of writing and it could be utterly irrelevant)

### Workspace affected

can't share, the id contains details about how we structure our repos

**Comment by pawlean** [comment]
Based on @trumbitta's message, I'm led to believe this is an IDE issue. However, @loujaybee / @akosyakov if you don't think it is please pass it on to the relevant team.

**Comment by trumbitta** [comment]
Just now I opened a workspace while a prebuild wasn't yet done, and the bug didn't show up

**Comment by akosyakov** [comment]
@mustard-mh @jeanp413 Could you investigate please? 🙏

**Comment by jeanp413** [comment]
This is duplicate of https://github.com/gitpod-io/gitpod/issues/7863, it's fixed on insiders, tomorrow Friday we will release stable 1.65 with the fix @trumbitta
Closing this so we track in the original issue
